server/src/main/java/com/paypal/selion/pojos/BrowserInformationCache.java

```diff
@@ -0,0 +1,242 @@
+            RemoteProxy remoteProxy = remoteProxyIterator.next();
+            urlList.add(remoteProxy.getRemoteHost());
+        }
+        return urlList;
+    }
+
+    private void removeIrrelevantURLs(List<URL> hotURLList) {
+        Iterator<URL> urlIterator = nodeMap.keySet().iterator();
+        while (urlIterator.hasNext()) {
+            URL cacheURL = urlIterator.next();
+            if (!hotURLList.contains(cacheURL)) {
+                nodeMap.remove(cacheURL);
+            }
+        }
+    }
+
+    /**
+     * <code>TestSlotInformation</code> holds the {@link BrowserInformation} corresponding to a individual unique test
+     * slot.
+     */
+    private static class TestSlotInformation {
+
+        private static final SeLionGridLogger logger = SeLionGridLogger.getLogger(TestSlotInformation.class);
+
+        // Browser information set.
+        private Set<BrowserInformation> browserInformationSet = null;
+
+        private TestSlotInformation() {
+            browserInformationSet = new HashSet<BrowserInformation>();
+        }
+
+        /**
+         * Updates the BrowserInformation. This method removes the existing entry and updates with the new entry.
+         * 
+         * @param browserInformation
+         *            Instance of {@link BrowserInformation}
+         * @return True if addition is successful, false othewise.
+         */
+        private boolean addBrowserInfo(BrowserInformation browserInformation) {
+            logger.entering(browserInformation);
+            if (!browserInformationSet.contains(browserInformation)) {
+                logger.log(Level.INFO, "Adding BrowserInfo " + browserInformation + " to set...");
+                return browserInformationSet.add(browserInformation);
+            } else {
+                logger.log(Level.INFO, "BrowserInfo " + browserInformation + " already present in set, replacing...");
+                browserInformationSet.remove(browserInformation);
+                return browserInformationSet.add(browserInformation);
+            }
+        }
+
+        private BrowserInformation getBrowserInfo(String browserName) {
+            logger.entering(browserName);
+            for (BrowserInformation browserInformation : browserInformationSet) {
+                if (browserInformation.getBrowserName().equals(browserName)) {
+                    logger.exiting(browserInformation);
+                    return browserInformation;
+                }
+            }
+            logger.log(Level.FINE, "requested browser name " + browserName + " unavailable in set... returning null");
+            return null;
+        }
+
+    }
+
+    /**
+     * Pojo for holding browser information
+     */
+    private static class BrowserInformation {
+
+        // Name of the browser
+        private final String browserName;
+
+        // Maximum instances
+        private final int maxInstances;
+
+        private volatile int hashCode;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field hashCode has the same name as a method](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103178711/issues/source?bid=2928425&fileBranchId=3143459#l191)